epochStart/metachain/triggerRegistry.go

```diff
@@ -27,50 +28,90 @@ func (t *trigger) LoadState(key []byte) error {
 
 	t.mutTrigger.Lock()
 	t.triggerStateKey = key
-	t.epochFinalityAttestingRound = state.EpochFinalityAttestingRound
-	t.currEpochStartRound = state.CurrEpochStartRound
-	t.prevEpochStartRound = state.PrevEpochStartRound
-	t.epoch = state.Epoch
-	t.epochStartMetaHash = state.EpochStartMetaHash
-	t.epochStartMeta = state.EpochStartMeta
+	t.epochFinalityAttestingRound = state.GetEpochFinalityAttestingRound()
+	t.currEpochStartRound = state.GetCurrEpochStartRound()
+	t.prevEpochStartRound = state.GetPrevEpochStartRound()
+	t.epoch = state.GetEpoch()
+	t.epochStartMetaHash = state.GetEpochStartMetaHash()
+	t.epochStartMeta = state.GetEpochStartMetaHeaderHandler()
+	t.epochChangeProposed = state.GetEpochChangeProposed()
 	t.mutTrigger.Unlock()
 
 	return nil
 }
 
 // UnmarshalTrigger unmarshalls the trigger with json, for backwards compatibility
-func UnmarshalTrigger(marshaller marshal.Marshalizer, data []byte) (*block.MetaTriggerRegistry, error) {
+func UnmarshalTrigger(marshaller marshal.Marshalizer, data []byte) (data.MetaTriggerRegistryHandler, error) {
+	trig, err := unmarshallTriggerV3(marshaller, data)
```

Comment:
could you rewrite this ? Like I do not like that in case of error we try the next one. that is not good.

Can you find some other way ? 

Is unmarshal failing only if the structure is wrong ?